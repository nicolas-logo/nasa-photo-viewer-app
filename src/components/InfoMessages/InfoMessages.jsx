// eslint-disable-next-line react/prop-types
export const InfoMessages = ({ apiErrorMessage, imagesLength, loading }) => (
    <div data-testid="info-messages">
        {apiErrorMessage !== null && <h4 className='text-danger'>{apiErrorMessage}</h4>}
        {imagesLength === 0 && !loading && <h4 className='text-warning'>No photos retrieved</h4>}
        {loading && <h4 className='text-white'>Loading...</h4>}
    </div>
)
