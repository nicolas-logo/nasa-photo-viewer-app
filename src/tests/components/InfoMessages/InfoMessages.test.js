import '@testing-library/jest-dom/extend-expect'
import { render, cleanup } from '@testing-library/react'
import { InfoMessages } from '../../../components/InfoMessages/InfoMessages'
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})

describe('InfoMessages tests', () => {
  it('should renders error message when `apiErrorMessage` prop is not null', () => {
    const apiErrorMessage = 'Error: Unable to retrieve photos.'
    const { getByText } = render(<InfoMessages apiErrorMessage={apiErrorMessage} />)
    const errorMessageElement = getByText(apiErrorMessage)
    expect(errorMessageElement).toBeInTheDocument()
    expect(errorMessageElement).toHaveClass('text-danger')
  })

  it('should renders "No photos retrieved" message when `imagesLength` is 0 and not loading', () => {
    const imagesLength = 0
    const loading = false
    const { getByText } = render(<InfoMessages imagesLength={imagesLength} loading={loading} />)
    const noPhotosMessage = getByText('No photos retrieved')
    expect(noPhotosMessage).toBeInTheDocument()
    expect(noPhotosMessage).toHaveClass('text-warning')
  })

  it('should renders "Loading..." message when `loading` prop is true', () => {
    const loading = true
    const { getByText } = render(<InfoMessages loading={loading} />)
    const loadingMessage = getByText('Loading...')
    expect(loadingMessage).toBeInTheDocument()
    expect(loadingMessage).toHaveClass('text-white')
  })

  it('should does not render any message when all props are null or false', () => {
    const { queryByText } = render(<InfoMessages />)
    expect(queryByText('Error: Unable to retrieve photos.')).toBeNull()
    expect(queryByText('No photos retrieved')).toBeNull()
    expect(queryByText('Loading...')).toBeNull()
  })

  it('should should match the snapshot', () => {
    const tree = renderer.create((<InfoMessages loading={false} imagesLength={0} />)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
