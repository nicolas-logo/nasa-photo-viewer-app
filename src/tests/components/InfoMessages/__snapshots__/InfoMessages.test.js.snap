// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoMessages tests should match the snapshot 1`] = `
<div
  data-testid="info-messages"
>
  <h4
    className="text-danger"
  />
  <h4
    className="text-warning"
  >
    No photos retrieved
  </h4>
</div>
`;

exports[`InfoMessages tests should should match the snapshot 1`] = `
<div
  data-testid="info-messages"
>
  <h4
    className="text-danger"
  />
  <h4
    className="text-warning"
  >
    No photos retrieved
  </h4>
</div>
`;
